src/rade_api.c

```diff
@@ -426,7 +457,7 @@ int rade_tx_eoo(struct rade *r, RADE_COMP tx_eoo_out[]) {
   return r->Neoo;
 }
 
-int rade_rx(struct rade *r, float features_out[], RADE_COMP rx_in[]) {
+int rade_rx(struct rade *r, float features_out[], int *has_eoo_out, float eoo_out[], RADE_COMP rx_in[]) {
```

Comment:
Just to make sure, `eoo_out` is basically structured so that odd indexed items are symbols and even indexed symbols are amplitudes, right? i.e.

eoo_out[0] = real value of symbol
eoo_out[1] = complex value of symbol
eoo_out[2] = amplitude
...

Reply:
Do the notes above `rade_rx()` in `rade_api.h` help?  

For the LDPC decoder you will need to calculate the symbol amplitudes.  As a starting point, I'd suggest making one estimate for the entire vector (e.g. the RMS amplitude across all symbols) and filling all entries of the amplitude[] vector with that value.